### PR TITLE
Simplify theme creation using CSS variables

### PR DESCRIPTION
**Pull request by TheGemDev**
#1918 
To simplify the creation of themes i would be creating only the following files which should contain all codes for the themes, the files are EventsSheet.css, Index.js, Markdown.css, Mosaic.css, Table.css(in the future we could merge Markdown.css, Mosaic.css, Table.css into a single file) 
Presently @4ian i have been able to use css variables for the tables.css in the themes(dark, default and nord theme) i would do the same for the solarized dark theme as soon as you view this and let me know what you think.

ps - sorry i added existing files

### CODE REVIEW

File: newIDE/app/src/UI/Theme/Theme/Table.css [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  --table-Selected-Row: #4c92ff;
+  --table-Header-Column-font-weight: 600;
+  --table-Header-Column-font-size: 12px;
+  --table-Header-Column-color: #9e9e9e;
+  --table-Header-Column-text-transform: none;
+  --table-Column-white-space: nowrap;
+  --table-Column-overflow: hidden;
+  --table-Column-text-overflow: ellipsis;
+}
+/*darktheme*/
```
Comment (4ian) [inline newIDE/app/src/UI/Theme/Theme/Table.css:24]: Each theme should have its own "Table.css" file (or a single css file, up to you), that contains just variables for the colors. Things like whitespace, margin, padding, overflow, font-size, etc... must be in a single CSS file used by all themes.

File: newIDE/app/src/UI/Theme/Theme/Table.css [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+.gd-table-default-theme .tableSelectedRow,
+.gd-table-dark-theme .tableSelectedRow {
+  background-color: var(--table-Selected-Row); /*OS X selection*/
+  color: var(--white);
+}
+
+.gd-table-default-theme .tableHeaderColumn,
+.gd-table-dark-theme .tableHeaderColumn {
```
Comment (4ian) [inline newIDE/app/src/UI/Theme/Theme/Table.css:81]: In the end, there should not be anymore different rule names (gd-table-default-theme, gd-table-dark-theme, gd-table-nord-theme...) for every theme.

There should be a single CSS rule (`gd-table`) that is used, and when you select a theme, only the variables setting the color are changed.

File: newIDE/app/src/UI/Theme/Theme/Table.css [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  text-transform: var(--table-Header-Column-text-transform);
+  font-weight: var(--table-Header-Column-font-weight);
+  font-size: var(--table-Header-Column-font-size);
+  color: var(--table-Header-Column-color);
```
Comment (4ian) [inline newIDE/app/src/UI/Theme/Theme/Table.css:85]: For example, this line is useful

File: newIDE/app/src/UI/Theme/Theme/Table.css [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+.gd-table-dark-theme .tableSelectedRow {
+  background-color: var(--table-Selected-Row); /*OS X selection*/
+  color: var(--white);
+}
+
+.gd-table-default-theme .tableHeaderColumn,
+.gd-table-dark-theme .tableHeaderColumn {
+  text-transform: var(--table-Header-Column-text-transform);
+  font-weight: var(--table-Header-Column-font-weight);
+  font-size: var(--table-Header-Column-font-size);
```
Comment (4ian) [inline newIDE/app/src/UI/Theme/Theme/Table.css:84]: For example, these lines should NOT be variables. Reconsider the idea behind this :)

**Comment by the-gem-dev** [comment]
@4ian so far i have only worked on the default theme(turns out my work on the dark theme had a lot of issues, so i am changing most of codes on the first commit based on your suggestion) and i would be awaiting your thoughts on it before it can be done for the currently existing themes.
Presently I have created a shared eventssheet.css, markdown.css, mosaic.css and table.css which all themes would use so therefore each theme would only have 3 files which are index.js, theme.css, Themecolors.css. so i would be awaiting your thoughts on it.

**Comment by the-gem-dev** [comment]
@4ian what do you think about this

**Comment by 4ian** [comment]
Can you explain what you've done? :) 
I still see a lot of similar `styles.css` files, while the goal is to only have one and themes are just specifying variables with colors?

**Comment by the-gem-dev** [comment]
i created a shared file for the events sheet, markdown, mosaic and table.css which all themes share. i tried creating one `styles.css` but it kept on picking the value of another css variable which belongs to another theme, now i think its the `:root` selector makes it a globally scoped variable.(i am not really sure) but i think so.

**Pull request closed without merging**